services/facade/server/models/account.js

```diff
@@ -14,36 +47,48 @@ module.exports = function(Account) {
     transactionService.getFunction = getFunction;
   });
 
-  function getFunction(model, method) {
-    let functionName = model + '_' + method;
-    return this.createModel(model, {})[functionName];
+  function findTransaction(input) {
+    let find = transactionService.getFunction('Transaction', 'findById');
+    return new Promise(function (resolve, reject) {
+      find(input, function(err, data) {
+        if (err) return reject(new Error(err));
+        resolve(data);
+      });
+    });
   }
 
-  Account.getAccountSummary = function(accountNumber, cb) {
-    let findCustomer = customerService.getFunction('Customer', 'findById');
-    let findAccount = accountService.getFunction('Account', 'findById');
-    let findAccountSummary = accountService.getFunction('AccountSummary', 'findById');
-    let findTransaction = transactionService.getFunction('Transaction', 'findById');
-    findAccountSummary({id: accountNumber, accountNumber: accountNumber}, function(err, accountSummary) {
-      if (err) return cb(err);
-      accountSummary = accountSummary.obj;
-      findAccount({id: accountNumber, accountNumber: accountNumber}, function(err, account) {
-        if (err) return cb(err);
-        findCustomer({id: account.obj.customerNumber, customerNumber: account.obj.customerNumber}, function(err, customer) {
-          if (err) return cb(err);
-          accountSummary.customer = customer.obj;
-          accountSummary.account = account.obj;
-          cb(null, accountSummary);
-        });
+  function findAccountSummary(input) {
+    let find = accountService.getFunction('AccountSummary', 'findById');
+    return new Promise(function (resolve, reject) {
+      find(input, function(err, data) {
+        if (err) return reject(new Error(err));
+        resolve(data);
       });
     });
   }
 
-  Account.listAllAccounts = function(customerNumber, cb) {
-    let findAccount = accountService.getFunction('Account', 'find');
-    findAccount({}, function(err, account) {
-      if (err) return cb(err);
-      cb(null, account.obj);
+  function findAccount(input) {
+    let find = accountService.getFunction('Account', 'findById');
+    return new Promise(function (resolve, reject) {
+      find(input, function(err, data) {
+        if (err) return reject(new Error(err));
+        resolve(data);
+      });
     });
-  };
+  }
+
+  function findCustomer(input) {
+    let find = customerService.getFunction('Customer', 'findById');
+    return new Promise(function (resolve, reject) {
+      find(input, function(err, data) {
+        if (err) return reject(new Error(err));
+        resolve(data);
+      });
+    });
+  }
+
+  function getFunction(model, method) {
+    let functionName = model + '_' + method;
+    return this.createModel(model, {})[functionName];
```

Comment:
IIUC, this is the single place where you are looking up callback-based functions. In which case this is the place where you can promisify the function before you return it back.

```js
return Promise.promisify(this.createModel(model, {})[functionName]);
```

However, this will not work for functions that expect `this` to be set. Here is a better solution:

```js
function getFunction(modelName, methodName) {
  const functionName = modelName + '_' + methodName;
  const model = this.createModel(modelName, {});
  const method = Promise.promisify(model[functionName]);
  return method.bind(model);
}
```